### ISSUE TITLE
SheenWoodLeatherSofa renders incorrectly

### ISSUE THREAD

**Issue by mrdoob** [gkjohnson/three-gpu-pathtracer#641]
**Describe the bug**

This new assets renders incorrectly:

https://github.com/KhronosGroup/glTF-Sample-Assets/pull/123

https://github.com/KhronosGroup/glTF-Sample-Assets/blob/e51c108880863c3d1ee161341a0d834a9152278c/Models/SheenWoodLeatherSofa/glTF-Binary/SheenWoodLeatherSofa.glb

**Screenshots and Repro Model**

* WebGLRenderer

![Screenshot 2024-05-23 at 10 48 05 AM](https://github.com/gkjohnson/three-gpu-pathtracer/assets/97088/3d41d279-2402-4e89-a2d7-4293a08930a3)

* PathtracerRenderer

![Screenshot 2024-05-23 at 10 47 49 AM](https://github.com/gkjohnson/three-gpu-pathtracer/assets/97088/6a023190-1b2d-4bd6-80bb-18e460a16b78)

**Comment by dongho-shin** [comment]
it looks fine in my mac

<img width="1008" alt="image" src="https://github.com/gkjohnson/three-gpu-pathtracer/assets/108255990/94ee680a-b122-45df-b450-e32ac41e94a4">

i try in three.js editor with royal_esplanade_1k.hdr background, environment

**Comment by gkjohnson** [comment]
@mrdoob - I'm not seeing the issue, either. Can you reproduce it? What kind of lighting setup were you using?

Here's what it looks like with some directional lights:

<img width="1013" alt="image" src="https://github.com/gkjohnson/three-gpu-pathtracer/assets/734200/c15ec888-7bf5-447c-ad7c-c9181282387b">

**Comment by dongho-shin** [comment]
I check in index(default example), it occasionally reproduces

**Comment by gkjohnson** [comment]
Here's the corrected file url that worked for me:

```
https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Assets/e51c108880863c3d1ee161341a0d834a9152278c/Models/SheenWoodLeatherSofa/glTF-Binary/SheenWoodLeatherSofa.glb
```

> I check in index(default example), it occasionally reproduces

Interesting - in addition to textures sometimes not working I'm seeing that an error is occasionally hit when building the merged geometry, which is odd. I wonder if there's some kind of loading race condition here? This will need a bit more digging.

**Comment by dongho-shin** [comment]
ah me too float32array fail when mergeGeometry called Float32array fail because of byteOffset and length over the lenth of arraybuffer
i'll dig it too

**Comment by dongho-shin** [comment]
FYI) currently i found something. mergeGeometries.js

```js
	// copy all the attribute data over
	const attributes = Object.keys( geometries[ 0 ].attributes );
	console.log('total Keys', attributes);
	for ( let i = 0, l = attributes.length; i < l; i ++ ) {

		let forceUpdateAttr = false;
		const key = attributes[ i ];
		if ( ! targetGeometry.getAttribute( key ) ) {

			console.log('has not key', key)
			const firstAttr = geometries[ 0 ].getAttribute( key );
			const attribute = createAttributeClone( firstAttr, totalAttributeCount );
			console.log('targetGeometry key', key, firstAttr, attribute, totalAttributeCount, geometries[0], geometries);
			targetGeometry.setAttribute( key, attribute );
			forceUpdateAttr = true;

		}
```
see this part
```js
      const firstAttr = geometries[ 0 ].getAttribute( key );
     const attribute = createAttributeClone( firstAttr, totalAttributeCount );
```
<img width="391" alt="image" src="https://github.com/gkjohnson/three-gpu-pathtracer/assets/108255990/732d0892-c9a3-4bb1-aa1e-c3df1bec5045">

index 0 BakeGeometry color attribute has item size 3 but index 1 bake geometry color attribute has item size 4 so it causes float32 array constructor range error

i'm going to find why BakeGeometry attribute itemSize is not same

**Comment by dongho-shin** [comment]
Almost BufferGeometries don't have color attributes.
However, this one does have a color attribute with an itemSize 3, which causes a problem. 
In the function called setCommonAttributes, if there is no colorAttribute, it defaults to white(1,1,1,1) with itemSize 4.
 What do you think about solution for this case,  the values that do not have itemSize 4 (maybe alpha) are filled with 1. 
 @gkjohnson

**Comment by gkjohnson** [comment]
@dongho-shin thank you very much for looking into this! I think that workaround looks good to me if you'd like to make a PR. Does this fix the texture rendering inconsistency, as well?

Very nice find!

**Comment by dongho-shin** [comment]
@gkjohnson In my case, fix texture rendering inconsistency yes if other edge case happened like geometry issue. let's open issue then. at least initialization fail was fixed I'll open pr for this

**Comment by gkjohnson** [comment]
Fixed in #657
